src/bitcoind.cpp

```diff
@@ -150,9 +159,20 @@ bool AppInit(int argc, char* argv[])
 
             pid_t sid = setsid();
             if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+                fprintf(stderr, "Error: setsid() failed: %s\n", strerror(errno));
+
+            (void)chdir("/");
```

Comment:
I like the chdir, myself, and think it will look neater than 'cd / && bitcoind' so may I propose two new command-line switches:

 * -chdir; CHange DIRectory to / (or C:\\) prior to daemonising, can perhaps later be changed into a noop.

 * -pwdir; Preserve Working DIRectory\*, currently a noop, can perhaps later be changed into preserving current functionality.

I considered giving -chdir an optional option to specify which directory, with / or C:\\ as the default, but that seems like overkill and makes option parsing Magic; a dirty word.

I think this fits well with bitcoind's triple life as a system daemon, user daemon or user interface.

----

[\*] Not what it really stands for.